source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -131,8 +133,8 @@ Additionally, Drivers that implement a Connection Pool MUST support the followin
       /**
        *  NOTE: This option has been deprecated in favor of timeoutMS.
        *
-       *  The maximum amount of time a thread can wait for a connection
-       *  to become available.
+       *  The maximum amount of time a thread can wait for either an available non-perished connection (limited by `maxPoolSize`),
+       *  or a pending connection (limited by `maxConnecting`).
```

Comment:
This clarifies the definition to match the reality. Needs to be done as part of this PR because this PR relies on the definition of `waitQueueTimeoutMS`.